hnn_core/network.py

```diff
@@ -869,20 +869,33 @@ def _instantiate_drives(self, n_trials=1):
             for drive in self.external_drives.values():
                 event_times = list()  # new list for each trial and drive
 
-                # loop over drive 'cells' and create event times for each
+                # loop over drives (one for each target cell population)
+                # and create event times
                 for this_cell_drive_conn in drive['conn'].values():
-                    for drive_cell_gid in this_cell_drive_conn['src_gids']:
-                        event_times.append(_drive_cell_event_times(
+                    if drive['cell_specific']:
+                        for drive_cell_gid in this_cell_drive_conn['src_gids']:
+                            event_times.append(_drive_cell_event_times(
+                                drive['type'], this_cell_drive_conn,
+                                drive['dynamics'], trial_idx=trial_idx,
+                                drive_cell_gid=drive_cell_gid,
+                                seedcore=drive['seedcore'])
+                            )
+                    else:
+                        # cell_specific=False should only have one src_gid
```

Comment:
I'm not sure these changes make sense, if `cell_specific=False` the events are never appended to `event_list`